src/std/iter.ss

```diff
@@ -119,27 +119,45 @@ package: std
           val))))
   (make-iterator port next))
 
-(def (iter-in-range start count step)
+(def (iter-in-iota start count step)
   (def (next it)
     (with ((iterator e) it)
       (with ([value . limit] e)
         (if (##fx> limit 0)
-          (let (value+step (+ value step))
-            (set! (car e) value+step)
+          (begin
+            (set! (car e) (+ value step))
             (set! (cdr e) (##fx- limit 1))
             value)
           iter-end))))
-  (if (and (number? start) (integer? count) (number? step))
+  (if (and (number? start) (fixnum? count) (number? step))
     (make-iterator (cons start count) next)
-    (error "Parameters are of wrong type (count:number start:integer step:number)."
+    (error "Parameters are of wrong type (count:fixnum start:number step:number)."
       [count start step])))
 
+(def* in-iota
+  ((count) (iter-in-iota 0 count 1))
+  ((count start) (iter-in-iota start count 1))
+  ((count start step) (iter-in-iota start count step)))
+
+(def (iter-in-range start end step)
+  (def (next it)
+    (with ((iterator e) it)
+      (if (< e end)
```

Comment:
This is buggy and doesn't match racket: `(for/collect (x (in-range 10 5 -2)) x) => ()`

Reply:
OK, so another patch is needed. Will you write it and update the documentation?